mono/metadata/threadpool-ms-io.c

```diff
@@ -62,23 +68,42 @@ enum {
 	AIO_OP_LAST
 };
 
+typedef enum {
+	UPDATE_EMPTY = 0,
+	UPDATE_ADD,
+	UPDATE_REMOVE_SOCKET,
+	UPDATE_REMOVE_DOMAIN,
+} ThreadPoolIOUpdateType;
+
 typedef struct {
 	gint fd;
 	MonoSocketAsyncResult *sockares;
-} ThreadPoolIOUpdate;
+} ThreadPoolIOUpdate_Add;
```

Comment:
This is broken, we're storing heap pointers without any safeguards. This will cause issues.